stock_lot_on_hand_first/models/stock_production_lot.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+from odoo.osv.expression import AND
+
+
+class ProductionLot(models.Model):
+    _inherit = "stock.production.lot"
+
+    @api.model
+    def _name_search(self, name="", args=None, operator="ilike", limit=80):
+        """Move lots without a qty on hand at the end of the list"""
+
+        if self.env.context.get("name_search_qty_on_hand_first"):
+            with_quantity_domain = AND([args, [("product_qty", ">", 0)]])
```

Comment:
`args` could be `None`, should we copy the line from parent method?
https://github.com/odoo/odoo/blob/120ca0804f90b0782ee5075213a77f9d4f72ceef/odoo/models.py#L1810

Reply:
Good catch